web/packages/teleterm/src/ui/services/resources/resourcesService.ts

```diff
@@ -73,14 +73,12 @@ export class ResourcesService {
   async searchResources({
     clusterUri,
     search,
-    filter,
+    filters,
     limit,
   }: {
     clusterUri: uri.ClusterUri;
     search: string;
-    // TODO(ravicious): Accept just `server | database | kube` as searchFilter here, wrap it in a
-    // variant of a discriminated union in searchResult.ts.
-    filter: ResourceTypeSearchFilter | undefined;
+    filters: SupportedResourceType[] | undefined;
```

Comment:
The `filter` field used to be of type `ResourceTypeSearchFilter | undefined` because we wanted to communicate that there might be no filter. In cases where an array is accepted, I think it's better to require the callsite to always provide an array and represent the "no filter" case by passing an empty array.